package com.google.android.gms.internal.ads;

import android.content.Context;
import java.util.concurrent.Executor;
import uf.o;

/* loaded from: classes3.dex */
public final class zzehd implements zzedc {
    private final Context zza;
    private final Executor zzb;
    private final zzdof zzc;

    public zzehd(Context context, Executor executor, zzdof zzdofVar) {
        this.zza = context;
        this.zzb = executor;
        this.zzc = zzdofVar;
    }

    @Override // com.google.android.gms.internal.ads.zzedc
    public final /* bridge */ /* synthetic */ Object zza(zzfca zzfcaVar, zzfbo zzfboVar, final zzecz zzeczVar) throws zzfcq, zzegu {
        zzdob zze = this.zzc.zze(new zzcrp(zzfcaVar, zzfboVar, zzeczVar.zza), new zzdoc(new zzdgc() { // from class: com.google.android.gms.internal.ads.zzehc
            @Override // com.google.android.gms.internal.ads.zzdgc
            public final void zza(boolean z11, Context context, zzcwg zzcwgVar) {
                zzecz zzeczVar2 = zzecz.this;
                try {
                    ((zzfdh) zzeczVar2.zzb).zzv(z11);
                    ((zzfdh) zzeczVar2.zzb).zzz(context);
                } catch (zzfcq e11) {
                    throw new zzdgb(e11.getCause());
                }
            }
        }));
        zze.zzd().zzo(new zzcma((zzfdh) zzeczVar.zzb), this.zzb);
        ((zzees) zzeczVar.zzc).zzc(zze.zzn());
        return zze.zzi();
    }

    @Override // com.google.android.gms.internal.ads.zzedc
    public final void zzb(zzfca zzfcaVar, zzfbo zzfboVar, zzecz zzeczVar) throws zzfcq {
        try {
            zzfcj zzfcjVar = zzfcaVar.zza.zza;
            if (zzfcjVar.zzo.zza == 3) {
                ((zzfdh) zzeczVar.zzb).zzr(this.zza, zzfcjVar.zzd, zzfboVar.zzv.toString(), (zzbpk) zzeczVar.zzc);
            } else {
                ((zzfdh) zzeczVar.zzb).zzq(this.zza, zzfcjVar.zzd, zzfboVar.zzv.toString(), (zzbpk) zzeczVar.zzc);
            }
        } catch (Exception e11) {
            o.h("Fail to load ad from adapter ".concat(String.valueOf(zzeczVar.zza)), e11);
        }
    }
}
